File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    # 딕셔너리로 name이랑 yearning을 묶어줌
    info = dict(zip(name, yearning))
    
    for i in photo:
        sum = 0
        for j in i:
            #딕셔너리에서 get메서드를 통해 값 가져옴
            
            sum += info.get(j, 0)
        answer.append(sum)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.swift
import Foundation

func solution(_ name:[String], _ yearning:[Int], _ photo:[[String]]) -> [Int] {
    //name이랑 yearning을 딕셔너리로 맵핑해야 함
    
    let wordToValue = Dictionary(uniqueKeysWithValues: zip(name, yearning))
    var temp: Int = 0
    var result: [Int] = []

    for i in photo {
        for j in i {
            // 만약 j가 딕셔너리에 존재한다면 -> keys로 바꿈
            if wordToValue.keys.contains(j) {
                if let value = wordToValue[j] {
                    temp += value
                }
            }
        }
        result.append(temp)
        temp = 0
    }
    return result
}